docs/rfcs/0001-designing-in-the-open.md

```diff
@@ -0,0 +1,139 @@
+---
+title: "Designing in the Open"
+weight: 1
+---
+
+# Designing in the Open
+
+* Date: 2021-11-02
+* Author: Robert Fratto (@rfratto)
+* PR: [grafana/agent#1055](https://github.com/grafana/agent/pull/1055)
+
+## Summary
+
+Traditionally, design proposals to Grafana Agent have been done in private by
+default, and then only made public after internal consensus was reached.
+This causes a few problems:
+
+* Community concerns are less likely to have an impact as core maintainers have
+  already agreed on a solution.
+* Initial design is gated to Grafana Labs employees.
+* Historical design docs with context and discussions become hard to find both
+  internally and even harder publicly.
+
+In the spirit of building a broader community, this document proposes inverting
+the design process to be public by default.
+
+## Goals
+
+* Outline a process to support public-first design documents
+* Lower the barrier to entry to becoming a maintainer that is not employed at
+  Grafana Labs
+* Encourage a community to contribute input to design documents
+* Encourage a community to contribute their own design documents
+
+## Non-Goals
+
+* Make everything public. If something can't be public for confidentiality
```

Comment:
This sentence has a lot of ambiguity. I would suggest removing "everything" and "something". Then define what should be confidential and what the scope is.  

I understand the goal here, but now (don't get me wrong, I know it's not the intention) it's a weird loophole.

Reply:
I'll rephrase that public by default should be best effort. 
I don't feel comfortable promising that everything will always be made public from the start (what if it's a security issue, etc.), and I'm not sure that any maintainer of this repository has authority to promise that either. 

The main problem I'm trying to address is how as a maintainer, I've personally been doing a lot of communication about the project internally first out of habit when I really didn't have to be doing that.

Reply:
I understand the goals and I really love what you want to achieve. Though it's just some nuances I guess. If we take the example of security; as a user and/or in the future contributor, it would please me to know how it's handled. 
Having a complete open project on multiple levels, yet with some parts confidential can become a bit weird. What if I get hit by a security issue and Grafana Labs knew this issue internally for a few weeks? 
I'm not saying everything must be public, though describing a few things on the how/why/what gives openness in a different form. If that makes any sense (trying hard here to describe my thoughts :) )

Reply:
Absolutely, and I do think it's worth thinking more about this. Describing things about openness more broadly as a mission statement sounds like it could work. 

> What if I get hit by a security issue and Grafana Labs knew this issue internally for a few weeks?

Yeah, gut feeling is that legal issues would prevent premature disclosure here. I think we do well w/r/t when we release public security issue postmortems, but I also believe that disclosing issues before there's an available fix and a CVE would also just put people more at risk.